File: java/querydsl/sample-jpa2/src/main/java/sample/model/Product.java
package sample.model;

import lombok.AccessLevel;
import lombok.Data;
import lombok.Setter;

import javax.persistence.*;
import java.math.BigDecimal;
import java.util.ArrayList;
import java.util.List;

@Data
@Entity
public class Product {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
	private String name;
	private BigDecimal price;

	@Setter(AccessLevel.NONE)
	@OneToMany(fetch = FetchType.EAGER, cascade= CascadeType.ALL)
	@JoinColumn(name = "product_id")
	private List<ProductVariation> variationList = new ArrayList<>();
}

File: blog/20150727/querydsl-scala-jpa-sample/src/main/scala-model/sample/model/Product.scala
package sample.model

import javax.persistence._

import java.util.ArrayList
import java.util.List
import java.math.BigDecimal

@Entity
class Product {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	var id: Long = _
	var name: String = _
	var price: BigDecimal = _

	@OneToMany(fetch = FetchType.EAGER, cascade= Array(CascadeType.ALL))
	@JoinColumn(name = "product_id")
	val variationList: List[ProductVariation] = new ArrayList()

	override def toString = s"Product(id: ${id}, name: ${name}, price: ${price}, variationList: ${variationList})"
}
